otelcol/factories.go

```diff
@@ -16,8 +16,14 @@ package otelcol // import "go.opentelemetry.io/collector/otelcol"
 
 import (
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
 )
 
 // Factories struct holds in a single type all component factories that
 // can be handled by the Config.
-type Factories = component.Factories // nolint:staticcheck
+type Factories struct {
+	component.Factories // nolint:staticcheck
+
+	// Connectors maps connector type names in the config to the respective factory.
+	Connectors map[component.Type]connector.Factory
+}
```

Comment:
We may have to wait to do this until next week. now `otelcol.Factories` is a different type than `component.Factories` so this is a breaking change for `otelcol.CollectorSettings.Factories` element.

Reply:
Alright, makes sense.

Reply:
This is unblocked if you rebase

Reply:
I've rebased to a tentative state, with the assumption that #6867 will provide the builder.